### DISCUSSION TITLE
canvas.toSVG clipPath only?

### DISCUSSION THREAD

**Discussion by ShuffledInk** [fabricjs/fabric.js#8801] [General]
Hello,

I'm trying to export only the clipPath area to SVG. Is there a way to do this? I know with toDataURL you can provide arguments for top, left, width, and height. I tried this with toSVG but no luck. Below is my code so far:

`const canvas = new fabric.Canvas("canvas", {
  backgroundColor: "#fff",
  originalWidth: 800,
  originalHeight: 600,
  width: 800,
  height: 600,
  controlsAboveOverlay: true
});

const canvasArea = new fabric.Rect({
  width: 600,
  height:500,
  stroke: "#000",
  fill: "#fff",
  objId: "main-canvas-area",
  selectable: false,
  evented: false,
})

canvas.add(canvasArea);
canvasArea.viewportCenterH();
canvasArea.viewportCenterV();
canvas.renderAll();

canvas.clipPath = canvasArea;

var svg = canvas.toSVG();
var link = document.createElement('a');
link.href = 'data:image/svg+xml,' + encodeURIComponent(svg);
link.download = 'clip-path.svg';

link.click();`

When this downloads, it is the full 800x600 instead of my clipPath 600x500

Appreciate any help!

**Comment by ShaMan123** [comment]
I don't understand why you would want this, what is the use case?
You can filter out object using intersection/geometry methods

**Reply by ShuffledInk** [reply]
I ran into an issue in my app where objects that are outside of canvas lose their controls. To resolve this I'm using a full-width canvas instead with a centered rect that clips everything outside. I need only the objects within the centered rect to export to png, jpeg or SVG.

**Reply by ShaMan123** [reply]
sounds like you are trying to workaround a workaround

**Reply by ShuffledInk** [reply]
If there's a better solution I'm open to it.

**Comment by ShaMan123** [comment]
Ohh you want to resize the viewbox.
Then do that.
